service/history/replication/executable_task.go

```diff
@@ -431,11 +438,71 @@ func (e *ExecutableTaskImpl) Resend(
 
 func (e *ExecutableTaskImpl) SyncState(
 	ctx context.Context,
-	remoteCluster string,
 	syncStateErr *serviceerrors.SyncState,
 	remainingAttempt int,
 ) (bool, error) {
-	return false, serviceerror.NewUnimplemented("not implemented") // TODO
+
+	remoteAdminClient, err := e.ClientBean.GetRemoteAdminClient(e.sourceClusterName)
+	if err != nil {
+		return false, err
+	}
+
+	targetClusterInfo := e.ClusterMetadata.GetAllClusterInfo()[e.ClusterMetadata.GetCurrentClusterName()]
+	_, err = remoteAdminClient.SyncWorkflowState(ctx, &adminservice.SyncWorkflowStateRequest{
+		NamespaceId: syncStateErr.NamespaceId,
+		Execution: &commonpb.WorkflowExecution{
+			WorkflowId: syncStateErr.WorkflowId,
+			RunId:      syncStateErr.RunId,
+		},
+		VersionedTransition: syncStateErr.VersionedTransition,
+		// TODO: add version histories to serviceerrors.SyncState and use it here.
+		// VersionHistories: nil,
+		TargetClusterId: int32(targetClusterInfo.InitialFailoverVersion),
+	})
+	if err != nil {
+		var syncStateErr *serviceerrors.SyncState
+		if !errors.As(err, &syncStateErr) {
```

Comment:
Do we need a specific error to indicate that transition history is disabled in source cluster? Or we just add task equivalents back to source cluster in all error cases?

Reply:
yeah I am even debating if SyncState is the right error type to use here. 

I will switch to FailedPrecondition type.